### ISSUE TITLE
Changing props.style from null to object fails as of react 0.13

### ISSUE THREAD

**Issue by skiano** [facebook/react#3424]
Hello,

I have a breaking change upgrading from 0.12 to 0.13. 

I have a component that uses inline styles, and in one state it uses inline styles and in the other it does not need them so i set style to null. In react 12 this worked fine, but now it fails to update the inline styles.

Here is an example of a component that isolates the issue:

```jsx
React.createClass({
  getInitialState: function () {
    return {
      isRed: false
    };
  },

  toggleRed: function () {
    this.setState({
      isRed: !this.state.isRed
    });
  },

  componentDidMount: function () {
    setInterval(this.toggleRed, 700);
  },

  render: function () {

    var style = null; // This does not work
    // var style = {}; // THIS WORKS

    if (this.state.isRed) {
      style = {
        color: 'red'
      };
    }

    return (
      <div style={style}>
        {this.state.isRed ? 'Is Red' : 'Is Black'}
      </div>
    );
  }
});
```

**Comment by andreypopp** [comment]
Looks like a dup of #3409 (fixed by #3410)

**Comment by skiano** [comment]
thanks!
